standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -13272,19 +13283,18 @@ public int deleteScheduledExecutions(int maxRetainSecs) {
       return 0;
     }
     boolean committed = false;
+    Query q = null;
     try {
       openTransaction();
       int maxCreateTime = (int) (System.currentTimeMillis() / 1000) - maxRetainSecs;
-      Query q = pm.newQuery(MScheduledExecution.class);
+      q = pm.newQuery(MScheduledExecution.class);
```

Comment:
The `Query` and `QueryWrapper` classes are `AutoCloseable` - which might be better than closing it by hand (`Query#close` calls `closeAll` so it does the same)

so I think it would be better to use try-with-resources to close these objects...at least in cases when there is already a `try` there...

Reply:
Done. Thanks for the review, @kgyrtkirk.  The try-with-resources exception handling written here may not be beautiful.